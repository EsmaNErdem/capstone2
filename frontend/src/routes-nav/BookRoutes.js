// Routes.js
import React from "react";
import { Route, Routes, Navigate  } from "react-router-dom";
import ProtectedRoute from "./ProtectedRoute";
import Home from "../homepage/Home";
import BookList from "../books/BookList";
import BookSearchList from "../books/BookSearchList";
import BookDetail from "../books/BookDetail";
import ReviewList from "../reviews/ReviewList";
import ReviewFilterList from "../reviews/ReviewFilterList";
import Profile from "../profile/Profile";
import LoginForm from "../profile/LoginForm";
import SignupForm from "../profile/SignupForm";

/** App routes.
 *
 * Parts of site should only be visitable when logged in. Those routes are
 * wrapped by <ProtectedRoute>, which is an authorization component.
 *
 * Visiting a non-existant route redirects to the homepage.
 */
const BookRoutes = ({ login, signup }) => {
  console.debug(
    "Routes",
    `login=${typeof login}`,
    `signup=${typeof signup}`,
  );

  return (
    <>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route element={<ProtectedRoute/>}>
          <Route path="/books" element={<BookList />} />
          <Route path="/books/search" element={<BookSearchList />} />
          <Route path="/books/:id" element={<BookDetail />} />
          <Route path="/reviews" element={<ReviewList />} />
          <Route path="/reviews/filter" element={<ReviewFilterList />} />
          <Route path="/profile/:username" element={<Profile />} />
        </Route>
        <Route path="/login" element={<LoginForm login={login} />} />
        <Route path="/signup" element={<SignupForm signup={signup} />} />
        <Route path="*" element={<Navigate to="/" />} />
      </Routes>
    </>
  );
};

export default BookRoutes;
