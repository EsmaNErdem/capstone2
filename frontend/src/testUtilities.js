import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  firstNsme: "testfirst",
  lastName: "testlast",
  email: "test@test.net",
  img: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasLikedReview = () => false, hasLikedBook = () => false, isUserReview = () => false, hasFollowing = () => false, setCurrentUser = u => u }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing, setCurrentUser, isUserReview }}>
      {children}
    </UserContext.Provider>
);

const CurrUserProvider =
    ({ children, currentUser = demoUser, hasLikedReview = () => false, hasLikedBook = () => false, isUserReview = () => true, hasFollowing = () => false, setCurrentUser = u => u }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing, setCurrentUser, isUserReview }}>
      {children}
    </UserContext.Provider>
);

const NonUserProvider =
    ({ children, currentUser = null, hasLikedReview = () => false, hasLikedBook = () => false, hasFollowing = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing,}}>
      {children}
    </UserContext.Provider>
);

export { UserProvider, CurrUserProvider, NonUserProvider };
