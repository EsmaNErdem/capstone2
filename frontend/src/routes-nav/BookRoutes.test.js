import { render, cleanup } from "@testing-library/react";
import BookRoutes from "./BookRoutes";
import { MemoryRouter } from "react-router";
import { UserProvider, NonUserProvider } from "../testUtilities";


jest.mock("../api", () => ({
  __esModule: true,
  default: {
    getUser: jest.fn(),
    registerUser: jest.fn(),
    loginUser: jest.fn(),
    unlikeReview: jest.fn(),
    likeReview: jest.fn(),
    sendBookReview: jest.fn(),
    deleteBookReview: jest.fn(),
    unlikeBook: jest.fn(),
    likeBook: jest.fn(),
    unfollowUser: jest.fn(),
    followUser: jest.fn(),
  },
}));

afterEach(() => {
  jest.clearAllMocks();
});

test("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <BookRoutes />
        </UserProvider>
      </MemoryRouter>,
  );
});

test("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <BookRoutes />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <NonUserProvider>
          <BookRoutes />
        </NonUserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
