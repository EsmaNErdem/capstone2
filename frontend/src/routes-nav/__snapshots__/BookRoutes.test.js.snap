// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      A Wrinkle in Page: Online Book Club
    </h1>
    <p>
      Join this social platform to find people like you who liove readint to talk about books
    </p>
    <h2>
      Welcome, testuser!
    </h2>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      A Wrinkle in Page: Online Book Club
    </h1>
    <p>
      Join this social platform to find people like you who liove readint to talk about books
    </p>
    <p>
      <a
        href="/login"
      >
        Login
      </a>
      <a
        href="/signup"
      >
        Sign Up
      </a>
    </p>
  </div>
</DocumentFragment>
`;
